### PR TITLE
core/fetcher: dedup aggregate attestations

### PR DESCRIPTION
**Pull request by dB2510**
Dedup aggregate attestations in fetcher for aggregators of same committee. This solves `clashing aggregated attestation` error in dutydb in case of different aggregate attestations for aggregators of same committee.

category: bug
ticket: none

### CODE REVIEW

File: core/fetcher/fetcher.go [hunk 1]
```diff
@@ -196,6 +199,11 @@ func (f *Fetcher) fetchAggregatorData(ctx context.Context, slot int64, defSet co
 		}
 		log.Info(ctx, "Resolved attester aggregation duty", z.Any("pubkey", pubkey))
 
+		_, ok = aggAttByCommIdx[attDef.CommitteeIndex]
+		if ok {
+			continue // Skips querying aggregate attestation for aggregators of same committee.
```
Comment (corverroos) [inline core/fetcher/fetcher.go:204]: please follow `fetchAttesterData` example, this will result in `resp[pubkey]` not being set for this validator

Comment (corverroos) [reply core/fetcher/fetcher.go:204]: please also add a test for this edge-case

**Review by corverroos — CHANGES_REQUESTED** [review]

File: core/fetcher/fetcher.go (hunk 1 above)
Comment (dB2510) [reply core/fetcher/fetcher.go]: Done!
Setting response for this validator also added tests for this case, converted existing tests to table tests.

**Review by corverroos — APPROVED** [review]
